src/http/session.rs

```diff
@@ -445,14 +477,29 @@ impl Stream for DefaultStream {
     }
 
     fn set_headers<'n, 'v>(&mut self, headers: Vec<Header<'n, 'v>>) {
-        self.headers = Some(headers.into_iter().map(|h| {
+        let mut new_headers = Vec::<Header<'static, 'static>>::new();
+
+        for h in headers.into_iter() {
```

Comment:
To make it perfect, you could avoid building this temporary `Vec` (avoiding the allocs/copies that it necessitates), by simply making `new_headers` an iterator that maps over the `into_iter` of the given `headers`:

```rust
let new_headers = headers.into_iter().map(|h| { /* transform h into the correct type as currently in the loop body */ });
```

Then, when you need the new `Vec` (in the `None` branch), you can do a `collect` on the iterator, while the `Some` could stay the same. This way, the original vector that the method got would be transformed directly into either the headers or would be expanded to extend it.

It's no biggie and if you don't feel like doing this, I'm fine with merging this, but iterators are very idiomatic Rust (for good reason, as they often make a lot of code more efficient) so I do recommend you keep them in mind going forward.